ocs_ci/deployment/baremetal.py

```diff
@@ -388,8 +389,26 @@ def destroy(self, log_level=''):
             """
             Destroy OCP cluster specific to BM UPI
             """
+
+            cluster_name = f"{constants.BM_DEFAULT_CLUSTER_NAME}.{config.ENV_DATA['base_domain']}"
+            zone_id = self.aws.get_hosted_zone_id(cluster_name=cluster_name)
+            result = self.aws.route53_client.list_resource_record_sets(HostedZoneId=zone_id)
+            logger.info("Checking if extra ip's are present in DNS record")
+            for node in self.mgmt_details:
+                if self.mgmt_details[node].get('extra_node'):
+                    for res in result['ResourceRecordSets']:
+                        if "apps" in res['Name']:
+                            for data in res['ResourceRecords']:
```

Comment:
Maybe instead of this for loop, we can replace it with a function with a name like `def find_extra_ips(resource_records)`? It's a little confusing to follow all the for loops